src/loki/search.cc

```diff
@@ -658,8 +658,15 @@ struct bin_handler_t {
         correlated.edges.erase(new_end, correlated.edges.end());
       }
 
+      //store filtered edges for retry in case find path with correlated.edges failed
+      for (auto filtered_edge : filtered) {
+          // we also add a panelty of 3600 to the score
+          filtered_edge.score = filtered_edge.score + 3600.0f;
```

Comment:
maybe this can happen in thor and instead of using a random number just always use the largest score from the edges which arent in the filtered set as the starting point?